src/connection/connection_v2.ts

```diff
@@ -122,6 +123,72 @@ export class ConnectionV2 extends BaseConnection {
     });
   }
 
+  async execute(
+    query: string,
+    executeQueryOptions: ExecuteQueryOptions = {}
+  ): Promise<Statement> {
+    if (this.options.useServerSidePreparedStatement) {
+      return this.executePreparedStatement(query, executeQueryOptions);
+    }
+    return super.execute(query, executeQueryOptions);
+  }
+
+  private async executePreparedStatement(
+    query: string,
+    executeQueryOptions: ExecuteQueryOptions
+  ): Promise<Statement> {
+    this.checkParameters(query, executeQueryOptions.namedParameters);
+    const { response } = await this.executeQuery(
+      query,
+      this.getExecuteQueryOptionsForPreparedStatement(executeQueryOptions)
+    );
+
+    const text = await response.text();
+    await this.throwErrorIfErrorBody(text, response);
+    return new Statement(this.context, {
+      query: query,
+      text,
+      executeQueryOptions
+    });
+  }
+
+  private checkParameters(
+    query: string,
+    namedParameters: Record<string, unknown> | undefined
+  ) {
+    const fireboltNumericParameters =
+      this.queryFormatter.getFireboltNumericParameters(query);
+    const providedParameters = Object.keys(namedParameters || {});
+    if (fireboltNumericParameters.length != providedParameters.length) {
+      throw new Error(
+        `Number of parameters in the query (${fireboltNumericParameters.length}) does not match the number of named parameters provided (${providedParameters.length}).`
+      );
+    }
+    fireboltNumericParameters.forEach(param => {
```

Comment:
I'm thinking if we even need to have this check? We will still fail on the server side if any parameter is missing. So we might as well simplify the logic here. But let's discuss